org.sonarlint.eclipse.core/src/org/sonarlint/eclipse/core/internal/preferences/SonarLintGlobalConfiguration.java

```diff
@@ -238,25 +239,37 @@ public static Collection<RuleConfig> readRulesConfig() {
     return deserializeRulesJson(json);
   }
 
+  private static class RuleConfigGson {
+
+    enum Level {
+      @SerializedName("on")
+      ON,
+      @SerializedName("off")
+      OFF
+    }
+
+    Level level = Level.ON;
+    Map<String, String> parameters;
```

Comment:
Could we initialise it with empty map and remove null check below?

Reply:
This will produce a different json:
```json
{
  "java:S123": {
    "level": "on",
    "parameters": {
    }
  }
}
```
vs
```json
{
  "java:S123": {
    "level": "on"
  }
}
```